packages/common/src/helpers/errors.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright (c) 2018-2021 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { AxiosError, AxiosResponse } from 'axios';
+import { HttpError } from "@kubernetes/client-node";
+
+/**
+ * This helper function does its best to get an error message from the provided object.
+ * @param error An object that supposed to contain an error message, or the error message.
+ * @returns Error message
+ */
+export function getMessage(error: unknown): string {
+  if (!error) {
+    return 'Unexpected error.';
+  }
+
+  if (isKubeClientError(error)) {
+    let statusCode = error.statusCode || error.response.statusCode;
+    if (!statusCode || statusCode === -1) {
+      return 'no response available due to network issue.';
+    }
+    if (error.body?.message) {
+      // body is from K8s in JSON form with message present
+      return error.body.message;
+    }
+    if (error.body) {
+      // pure http response body without message available
+      return error.body;
+    }
+    return `"${statusCode}" returned by "${error.response.url}".`;
+  }
+
+  if (isAxiosError(error) && isAxiosResponse(error.response)) {
+    const response = error.response;
+    if (response.data.message) {
+      return response.data.message;
+    } else if (response.config.url) {
+      return `"${response.status} ${response.statusText}" returned by "${response.config.url}".`;
+    } else {
+      return `"${response.status} ${response.statusText}".`;
+    }
+  }
+
+  if (isErrorLike(error)) {
+    return error.message;
+  }
+
```

Comment:
These changes without supporting type 'string' as an Error bring a lot of wrong messages.

For example 'updateWorkspace'
The first time it is using in storage

```
  updateWorkspace: (workspace: che.Workspace): 
...
    try {
      const updatedWorkspace = await cheWorkspaceClient.restApiClient.update(workspace.id, workspace);
...
    } catch (e) {
const errorMessage = `Failed to update the workspace ${workspace.devfile.metadata.name}, reason: ` + common.helpers.errors.getMessage(e);
...
```
https://github.com/eclipse-che/che-dashboard/pull/329/files#diff-3c70d20df3faae24a13278f6cc9c832b435c183fedfeb19aea49575d56420eefR340

 The second time in the controller
```
...    
try {
      await this.props.updateWorkspace(changedWorkspace);
...
    } catch (e) {
      const errorMessage = common.helpers.errors.getMessage(e);
...

```
https://github.com/eclipse-che/che-dashboard/pull/329/files#diff-34b89a5b5c3edd3273d30d794ff491e5abd7e9c710f70b5a68a119a6383bade8R115

The resulting message always will be the same:
'Unexpected error. Check DevTools console and network tabs for more information.'

Reply:
Nice catch! Thanks for pointing this out cause I overlooked this case.